### PR TITLE
Optionally propagate the exit code

### PR DESCRIPTION
**Pull request by hnykda**
Hi,

I added an `--exit-code` option which tries to propagate the exit code of the underlying program execution. `mprof` is just silently eating it, causing some unexpected behavior when used in e.g. scripts, containers, ...

It is backward compatible in a way that you need to pass `--exit-code|-E` option in the `run` option.

One disruptive change is that I added logger from the standard `logging` library + added the basicConfig. This is not necessary but IMO is a very good standard in the python world. `print` are not respecting formatters etc., I consider their usage wrong.

### CODE REVIEW

File: mprof.py [hunk 1]
```diff
@@ -197,7 +202,7 @@ def run_action():
                              'Option 4: (--python flag present) "<PYTHON_MODULE> <ARG1> <ARG2>..." - profile python module\n'
                         )
     args = parser.parse_args()
-
+    print(args)
```
[the remark this answers is not in the document]
Comment (hnykda) [reply mprof.py]: It is, sorry, removed

**Comment by fabianp** [comment]
thanks @hnykda for the contribution. Changes look good to me

**Comment by hnykda** [comment]
We are using `mprof` for long-running processing job, where failures are common (e.g. dataset couldn't be processed or the dynamic data got from a 3rd party malformed). So it's part of a longer pipeline. We collect these log files and can historically compare if we are getting better or worse with memory consumption.

**Comment by fabianp** [comment]
Regarding printing vs logging, I think logging adds some flexibility. I don't have a use for it but also don't see any reason not to use it. Hence I'm fine with it.

**Comment by fabianp** [comment]
@Juanlu001 : are you OK with the changes in this PR?

**Comment by Juanlu001** [comment]
LGTM!

**Pull request merged**